### ISSUE TITLE
Enforce attachment media sizing limits more strictly

### ISSUE THREAD

**Issue by michaelkirk** [signalapp/Signal-iOS#1233]
We want to enforce some hard file size limits on the media we let users attach.

At the moment we're enforcing a max resolution and re-compressing it without verifying the actual size.

When a too-large GIF is sent, we recompress it as JPG (as of https://github.com/WhisperSystems/Signal-iOS/pull/1232) rather than just failing. Ideally we could do some nice resizing to get it within our limits without losing the animation.

Here are the limits we want to enforce (lifted from [Signal-Android](https://github.com/WhisperSystems/Signal-Android/blob/master/src/org/thoughtcrime/securesms/mms/PushMediaConstraints.java)).

GifMaxSize return 5 \* MB;
ImageMaxSize return 420 \* KB;
VideoMaxSize return 100 \* MB;
getAudioMaxSize 100 \* MB; (Signal-iOS doesn't currently allow audio recording #230 )

**Comment by caveman1973** [comment]
- What about giving two settings? One for Wi-Fi and another for cellular.

The actual compression for video is horrible. It compress too much.

**Comment by devurandom** [comment]
#2428 might be a duplicate.

**Comment by hyiltiz** [comment]
Enforcing an upper limit for media could have consequences for some users (e.g. journalists) who rely on the privacy offered by Signal to safely transfer media; they might be forced to split the media outside Signal (not safe), or worse, transfer it through a platform without the limit.

Instead, all platforms should allow "media management", so any user should be allowed to sort large attachments in their device, and permanently delete them. This feature is offered only in the Android version as of Dec, 2021.

**Comment by hyiltiz** [comment]
Dedication from the devs is blocking it; stale bot please draw attention here.

**Comment by hyiltiz** [comment]
Stale bot, please.

**Comment by hyiltiz** [comment]
@michaelkirk Has there been PRs merged in response to this? Or are we closing out of frustration?

**Comment by michaelkirk** [comment]
When I created this issue, it was about being smarter about how Signal-iOS recompresses media it is sending. This was addressed in a few different ways, long ago.

My understanding of your comments in this thread, are that you are interested in a different media issue — managing the media stored on your device. If you want to discuss that feature, the forum would be a better place to brainstorm with folks: https://community.signalusers.org/c/feature-requests/ios-feature-requests/19
